app/services/segment_delete_service.rb

```diff
@@ -0,0 +1,88 @@
+# frozen_string_literal: true
+
+class SegmentDeleteService
+  SEGMENT_URI = URI('https://gdpr.segment.com/graphql')
+
+  class SegmentDeleteServiceError < StandardError
+    def initialize(message:, response:)
+      super(message)
+      @response = {body: response.body, code: response.code}
+    end
+
+    attr_reader :response
+
+    def error_type
+      :delete_service
+    end
+  end
+  class SegmentAuthenticationError < SegmentDeleteServiceError
```

Comment:
Looks strange to me to inherit an authentication error to a method error (deletion of a service)
They might share the same base but do not have the same meaning

Reply:
It didn't mean to be a method error but a 'error of the segment service'. Anyway, this won't exist anymore in the new version I think 😄